### PR TITLE
Add timeout to wait_ready

### PR DESCRIPTION
**Pull request by grob6000**
Add a 5 secondish timeout to wait_ready.
Throws a TimeoutError exception if the device does not respond.
As per #36, it's currently not possible to determine whether the device is present except for attempting to communicate and receiving no ready signal. Some users will want to know whether the device is connected or not, e.g.:
- Applications that support multiple screen types
- In case of screen fault, so the script doesn't hang completely (an exception is more statesmanly response; applications can't detect the infinite loop)

Applications can catch the TimeoutError exception when initializing the display and use this to determine whether the device is connected; e.g. from mine:

```
  def connect(self):
    try:
      self._epddisplay = AutoEPDDisplay(vcom=self.vcom)
    except IOError:
      logger.warning("EPD Display failed to init; will not function")
      self._epddisplay = None   
```

### CODE REVIEW

File: IT8951/spi.pyx [hunk 1]
```diff
@@ -99,8 +99,11 @@ cdef class SPI:
         '''
         Wait for the device's ready pin to be set
         '''
-        while not GPIO.input(Pins.HRDY):
-            sleep(0.001)
+        for i in range(0,5000):
```
Comment (txoof) [inline IT8951/spi.pyx:102]: this is a bit more elegant than what I proposed in #36.

**Comment by GregDMeyer** [comment]
I made some small changes and committed it! I don't think I have push access to your fork so I did it in another commit (a8df9840394b627f5f05870340ab88476fda4a48). Thanks for the PR! :-)

**Comment by GregDMeyer** [comment]
Oh one quick question---I realized I changed it to TimeoutError but you had it as IOError. Is there a reason you feel IOError is more appropriate in this setting?

**Comment by grob6000** [comment]
No particular reason; as long as it raises up to let the user handle it
should be fine!

Thanks!!

**Pull request closed without merging**